File: Codes/Hash/Non-Cryptographic/ELFHash/ELFHash.rb
# Modifikasi PJWHash untuk sistem UNIX.
# Archive of Reversing.ID
# Non-Cryptographic Hash

def elfhash( key, len=key.length )
    state = 0
    x = 0
    len.times{ |i|
        state = (state << 4) + key[i]
        if  (x = state & 0xF0000000) != 0
            state ^= (x >> 24)
            state &= ~x
        end
    }
    return state
end

File: Codes/Hash/Non-Cryptographic/ELFHash/ELFHash.py
# Modifikasi PJWHash untuk sistem UNIX.
# Archive of Reversing.ID
# Non-Cryptographic Hash

def ELFHash(key):
    state = 0
    test  = 0

    for i in range(len(key)):
        state = (state << 4) + ord(key[i])
        
        test = state & 0xF0000000
        if test != 0:
            state ^= (test >> 24)
        
        state &= ~test
    
    return state